testnet/stacks-node/src/syncctl.rs

```diff
@@ -80,11 +80,18 @@ impl PoxSyncWatchdogComms {
     }
 
     /// Wait for at least one inv-sync state-machine passes
-    pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> bool {
+    pub fn wait_for_inv_sync_pass(
```

Comment:
A couple things:

* You'll need a similar check here for `wait_for_download_pass`.
* I think a copy of the `should_keep_running` Arc should be added as a member of the `PoxSyncWatchdogComms` struct.  Then you won't need to pass it as an argument.
* You could add a helper method that checks if the `should_keep_running` flag has been set, so you can just add the check into the `while` loop's condition.

Reply:
Actually, a really economical way to do this would be to add a custom `interruptable_sleep()` method, like so:

```
fn interruptable_sleep(&self, secs: u64) -> Result<(), burnchain_error> {
   let deadline = secs + get_epoch_time_secs();
   while get_epoch_time_secs() < deadline {
      sleep_ms(1000);
      if !self.should_keep_running.load(Ordering::SeqCst) {
         return Err(burnchain_error::CoordinatorClosed);
      }
   }
   Ok(())
}
```

Then, you could replace all instances of `sleep_ms(TIME)` in this module with `self.interruptable_sleep(TIME)?`, and return an error to the caller if the node has been terminated.  If the caller gets a `CoordinatorClosed` error, then it should also stop whatever it was doing and propagate that error, all the way up to the main loop.

Reply:
Addressed with latest